text/0043-better-eviction.md

```diff
@@ -0,0 +1,126 @@
+# Improvements to faulty validator handling
+
+## Summary
+
+[summary]: #summary
+
+CEP PR: [casperlabs/ceps#0043](https://github.com/casperlabs/ceps/pull/0043)
+
+Exclude validators that equivocated or were inactive in one era from all future eras,
+even before the `auction_delay`.
+
+
+## Motivation
+
+[motivation]: #motivation
+
+"Official" removal of a faulty (i.e. inactive or equivocating) validator from the validator set
+happens with a delay:
+
+In Highway, validators take turns at proposing blocks, and the sequence of proposers is determined
+pseudorandomly. It is important that an attacker cannot manipulate the system in a way that allows
+them to choose among a large number of proposer sequences and thus get more slots. That is why
+the validator weights are determined in advance, with an `auction_delay` between the auction and
+the beginning of the era: A small change to any weight would change the proposer sequence for
+everyone, so it is important that before the auction, while bids can still change, the PRNG seed is
+not yet known.
+
+After the auction, during that delay, every block adds a single bit to the random seed that
+will be used to initialize the PRNG. That way, an attacker, even if they had the last three blocks
+before the new era's beginning, can choose only among eight different seeds, which are
+all likely to be fair. If they could still e.g. reduce two bids with a million tokens each, they
+could choose among a trillion different proposer sequences. Similarly, if an attacker has k
+validator nodes and could remove themselves by letting any subset of nodes exhibit a fault,
+they could choose among 2<sup>k</sup> leader sequences, even if those k nodes' stakes are small.
```

Comment:
```suggestion
they could choose among 2ᵏ leader sequences, even if those k nodes' stakes are small.
```

Reply:
It does render correctly on GitHub, though.